gordon/plugins_loader.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Module for loading plugins distributed via third-party packages.
+
+Plugin discovery is done via ``entry_points`` defined in a package's
+``setup.py``, registered under ``'gordon.plugins'``. For example:
+
+.. code-block:: python
+
+    # setup.py
+    from setuptools import setup
+
+    setup(
+        name=NAME,
+        # snip
+        entry_points={
+            'gordon.plugins': [
+                'gcp.gpubsub = gordon_gcp.gpubsub:EventClient',
+                'gcp.gce.a = gordon_gcp.gce.a:ReferenceSourceClient',
+                'gcp.gce.b = gordon_gcp.gce.b:ReferenceSourceClient',
+                'gcp.gdns = gordon_gcp.gdns:DNSProviderClient',
+            ],
+        },
+        # snip
+    )
+
+Plugins are initialized with any config defined in ``gordon-user.toml``
+and ``gordon.toml``. See :doc:`config` for more details.
```

Comment:
`:doc:` will actually insert a link to the config page (`gordon/docs/config.rst`).